specification/cognitiveservices/data-plane/AnomalyDetector/stable/v1.1/cadl/multivariate/models.cadl

```diff
@@ -0,0 +1,370 @@
+10.
+""")
+  topContributorCount: int32;
+
+  @doc("""
+A required field, indicating the start time of data for detection, which should
+be date-time of ISO 8601 format.
+""")
+  startTime: zonedDateTime;
+
+  @doc("""
+A required field, indicating the end time of data for detection, which should
+be date-time of ISO 8601 format.
+""")
+  endTime: zonedDateTime;
+}
+
+model AnomalyState {
+  @doc("The timestamp for this anomaly.")
+  timestamp: zonedDateTime;
+  value?: AnomalyValue;
+
+  @doc("Error message for the current timestamp.")
+  errors?: ErrorResponse[];
+}
+
+model AnomalyValue {
+  @doc("True if an anomaly is detected at the current timestamp.")
+  isAnomaly: boolean;
+
+  @doc("""
+Indicates the significance of the anomaly. The higher the severity, the more
+significant the anomaly is.
+""")
+  @minValue(0.0)
+  @maxValue(1.0)
+  severity: float32;
+
+  @doc("""
+Raw anomaly score of severity, will help indicate the degree of abnormality as
+well.
+""")
+  @minValue(0.0)
+  @maxValue(2.0)
+  score: float32;
+  interpretation?: AnomalyInterpretation[];
+}
+
+model AnomalyInterpretation {
+  @doc("Variable.")
+  variable?: string;
+
+  @doc("""
+This score shows the percentage contributing to the anomalous timestamp. A
+number between 0 and 1.
+""")
+  contributionScore?: float32;
+  correlationChanges?: CorrelationChanges;
+}
+
+model CorrelationChanges {
+  @doc("The correlated variables that have correlation changes under an anomaly.")
+  changedVariables?: string[];
+}
+
+@doc("""
+Training result of a model including its status, errors and diagnostics
+information.
+""")
+model ModelInfo {
+  @doc("""
+Source link to the input data to indicate an accessible Azure storage Uri,
+either pointed to an Azure blob storage folder, or pointed to a CSV file in
+Azure blob storage based on you data schema selection. 
+""")
+  dataSource: string;
+
+  @doc("""
+Data schema of input data source: OneTable or MultiTable. The default
+DataSchema is OneTable.
+""")
+  dataSchema?: DataSchema;
+
+  @doc("""
+A required field, indicating the start time of training data, which should be
+date-time of ISO 8601 format.
+""")
+  startTime: zonedDateTime;
+
+  @doc("""
+A required field, indicating the end time of training data, which should be
+date-time of ISO 8601 format.
+""")
+  endTime: zonedDateTime;
+
+  @doc("""
+An optional field. The display name of the model whose maximum length is 24
+characters.
+""")
+  @maxLength(24)
+  displayName?: string;
+
+  @doc("""
+An optional field, indicating how many previous timestamps will be used to
+detect whether the timestamp is anomaly or not. 
+""")
+  slidingWindow?: int32;
+
+  @doc("An optional field, indicating the manner to align multiple variables.")
+  alignPolicy?: AlignPolicy;
+
+  @doc("Model status. One of CREATED, RUNNING, READY, and FAILED.")
+  status?: ModelStatus;
+
+  @doc("Error messages when failed to create a model.")
+  errors?: ErrorResponse[];
```

Comment:
Java apiview: is it read-only?

ModelInfo is used in both input and output. Hence all its read-only properties need to be specified correctly.

Reply:
The read-only properties are the ones that shows "required":true in our [swagger file](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/AnomalyDetector/stable/v1.1/MultivariateAnomalyDetector.json).

Reply:
@jr-MS: errors should be read only, please confirm if that is correct. ModelInfo is an input+output model.

Reply:
"status" might be read-only as well. But only service would be sure about them.

@jr-MS I don't think they are same. From your sample they are apparently not read-only https://github.com/Azure/azure-sdk-for-java/blob/31de7784741d23aac3ccc992dae11ac34adf5b76/sdk/anomalydetector/azure-ai-anomalydetector/src/samples/java/com/azure/ai/anomalydetector/MultivariateSample.java#L120

Reply:
@jr-MS An issue should be added to check+update ModelInfo since it's an input/output model and fields should be correctly updated based on whether they will only be returned as part of the output. I wouldn't block merging this PR or the preview on this however.

Reply:
Thanks @catalinaperalta , I've added this in backlog adn add check+update ModelInfo for next iteration. @minkefusiji for awareness as well.